### PR TITLE
Migrate iterator query using scan entries for better performance

### LINKED ISSUE #1273: Migrate DB from MVCC to EntryCatalog is slow
Listing entries from MVCC using paging API is very slow and while migrate does it for each commit, migrate can it a very long time.

### PR DESCRIPTION
**Pull request by nopcoder**
Close #1273

### CODE REVIEW

File: catalog/migrate/iterator.go [hunk 1]
```diff
@@ -3,74 +3,69 @@ package migrate
...
+	return &Iterator{
+		rows: rows,
+	}, nil
 }
 
 func (it *Iterator) Next() bool {
-	if it.err != nil {
+	if !it.rows.Next() {
+		it.err = it.Err()
+		it.value = nil
```
Comment (guy-har) [inline catalog/migrate/iterator.go:43]:
```suggestion
	if !it.rows.Next() {
		it.err = it.rows.Err()
		it.value = nil
```

File: catalog/migrate/iterator.go [hunk 2]
```diff
@@ -81,46 +76,16 @@ func (it *Iterator) Value() *rocks.EntryRecord {
 }
 
 func (it *Iterator) Err() error {
-	return it.err
+	return it.rows.Err()
```
Comment (guy-har) [inline catalog/migrate/iterator.go:79]:
```suggestion
	return it.err
```

File: catalog/migrate/iterator.go [hunk 3]
```diff
@@ -3,74 +3,69 @@ package migrate
...
+	if it.err != nil {
 		return
 	}
-	it.state = iteratorStateInit
-	it.offset = id.String()
-	it.buf = it.buf[:0]
-	it.value = nil
-	it.err = nil
+	it.rows.Close()
+	it.err = ErrNotSeekable
```
Comment (guy-har) [inline catalog/migrate/iterator.go:68]: You don't have an error check on Next, not sure what happens if you call next after close

**Review by guy-har — CHANGES_REQUESTED** [review]
Looks good,
seems like there is a small problem with the error in the migrate iterator

File: catalog/migrate/iterator.go (hunk 3 above)
Comment (nopcoder) [reply catalog/migrate/iterator.go:68]: added

File: catalog/migrate/iterator.go (hunk 2 above)
Comment (nopcoder) [reply catalog/migrate/iterator.go:79]: fixed - part of moving from using iterator error my our own err.

**Review by guy-har — APPROVED** [review]
Cool, Thanks

**Comment by nopcoder** [comment]
> Cool, Thanks

Thank you!

**Pull request merged**